src/main/java/stsjorbsmod/effects/CampfireThirstEffect.java

```diff
@@ -0,0 +1,110 @@
+package stsjorbsmod.effects;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.Interpolation;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.relics.AbstractRelic;
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+import com.megacrit.cardcrawl.rooms.CampfireUI;
+import com.megacrit.cardcrawl.rooms.RestRoom;
+import com.megacrit.cardcrawl.vfx.AbstractGameEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardBrieflyEffect;
+import stsjorbsmod.patches.CampfireThirstPatch;
+import stsjorbsmod.patches.WrathField;
+
+import java.util.Iterator;
+
+import static stsjorbsmod.JorbsMod.makeID;
+
+public class CampfireThirstEffect extends AbstractGameEffect {
+    public static final String[] TEXT;
+    private Color screenColor;
+    private boolean openedScreen = false;
+    private int thirstHeal;
+    private int thirstMaxHPIncrease;
+
+    public CampfireThirstEffect() {
+        this.screenColor = AbstractDungeon.fadeColor.cpy();
+        this.duration = 1.5F;
+        this.screenColor.a = 0.0F;
+        AbstractDungeon.overlayMenu.proceedButton.hide();
+
+        // Did this in the constructor rather than the update in case there are relics
+        // for Cull that tweak these numbers.
+        this.thirstHeal = 5;
+        this.thirstMaxHPIncrease = 1;
+    }
+
+    public void render(SpriteBatch sb) {
+        sb.setColor(this.screenColor);
+        sb.draw(ImageMaster.WHITE_SQUARE_IMG, 0.0F, 0.0F, (float) Settings.WIDTH, (float)Settings.HEIGHT);
+        if (AbstractDungeon.screen == AbstractDungeon.CurrentScreen.GRID) {
+            AbstractDungeon.gridSelectScreen.render(sb);
+        }
+    }
+
+    public void update() {
+        if (!AbstractDungeon.isScreenUp) {
+            this.duration -= Gdx.graphics.getDeltaTime();
+            this.updateBlackScreenColor();
+        }
+
+        Iterator var1;
+        if (!AbstractDungeon.isScreenUp && !AbstractDungeon.gridSelectScreen.selectedCards.isEmpty()) {
+            var1 = AbstractDungeon.gridSelectScreen.selectedCards.iterator();
+
+            while(var1.hasNext()) {
```

Comment:
Can simplify a few lines of iterator stuff here down to `for (AbstractCard c : AbstractDungeon.gridSelectScreen.selectedCards) {`

I'm guessing this was probably copied from IntelliJ's decompilation of `CampfireSmithEffect` - generally, if you're using a decompiled base game class as a starting point, I'd recommend using [JD-GUI](https://java-decompiler.github.io/) to do the decompilation instead of trusting IntelliJ; IntelliJ is convenient to F12 with, but JD-GUI has more accurate and more conventional output (it uses a for-each loop in this spot in `CampfireSmithEffect`, for example)